src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/RazorDocumentOptionsService.cs

```diff
@@ -0,0 +1,80 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.ComponentModel.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.ExternalAccess.Razor;
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.Razor.Editor;
+
+namespace Microsoft.VisualStudio.LanguageServerClient.Razor
+{
+    [Export(typeof(IRazorDocumentOptionsService))]
+    internal sealed class RazorDocumentOptionsService : IRazorDocumentOptionsService
+    {
+        private readonly RazorLSPClientOptionsMonitor _optionsMonitor;
+
+        [ImportingConstructor]
+        public RazorDocumentOptionsService(RazorLSPClientOptionsMonitor optionsMonitor)
+        {
+            if (optionsMonitor is null)
+            {
+                throw new ArgumentNullException(nameof(optionsMonitor));
+            }
+
+            _optionsMonitor = optionsMonitor;
+        }
+
+        public Task<IRazorDocumentOptions> GetOptionsForDocumentAsync(Document document, CancellationToken cancellationToken)
```

Comment:
If there's an easy way to unit test this and its response types I'd do it. If not I wouldn't worry about it.

Reply:
Added! I ended up having to borrow some code from the LanguageServices layer to initialize the Roslyn doc, but I was able to get things working in the end.